lib/rspec/mocks/message_expectation.rb

```diff
@@ -650,7 +650,7 @@ def call(*_args_to_ignore, &block)
             @error_generator.raise_wrong_arity_error(args, block_signature)
           end
 
-          value = @eval_context ? @eval_context.instance_exec(*args, &block) : block.call(*args)
+          value = @eval_context ? @eval_context.instance_exec(*args, &block) : yield(*args)
```

Comment:
Does it make sense to remove the `&block` from the signature since you've switched to `yield`?

Reply:
this one is a little weird since `&block` is still used earlier in the line.

Reply:
oh yeah, that is kinda weird.